packages/core/snapshot/write.go

```diff
@@ -54,76 +61,79 @@ func StreamSnapshotDataTo(
 			// write rests of outputWithMetadatas
 			err = writeOutputWithMetadatasFunc(chunksOutputWithMetadata)
 			if err != nil {
-				return err
+				return nil, err
 			}
 			break
 		}
 
 		outputWithMetadataCounter++
+		outputChunkCounter++
 		chunksOutputWithMetadata = append(chunksOutputWithMetadata, output)
 
-		// put a delimeter every 100 outputs
-		if outputChunkCounter == 100 {
+		// put a delimeter every utxoStatesChunkSize outputs
+		if outputChunkCounter == utxoStatesChunkSize {
 			err = writeOutputWithMetadatasFunc(chunksOutputWithMetadata)
 			if err != nil {
-				return err
+				return nil, err
 			}
 			chunksOutputWithMetadata = make([]*ledger.OutputWithMetadata, 0)
+			outputChunkCounter = 0
 		}
 	}
 
 	// write epochDiffs
 	epochDiffs, err := epochDiffsProd()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	typeSet := new(serix.TypeSettings)
-	bytes, err := serix.DefaultAPI.Encode(context.Background(), epochDiffs, serix.WithTypeSettings(typeSet.WithLengthPrefixType(serix.LengthPrefixTypeAsUint32)), serix.WithValidation())
+	bytes, err := serix.DefaultAPI.Encode(context.Background(), epochDiffs, serix.WithValidation())
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if err := writeFunc(fmt.Sprintf("diffEpoch"), append(bytes, delimiter...)); err != nil {
-		return err
+		return nil, err
 	}
 
 	// seek back to the file position of the outputWithMetadata counter
 	if _, err := writeSeeker.Seek(0, io.SeekStart); err != nil {
-		return fmt.Errorf("unable to seek to LS counter placeholders: %w", err)
+		return nil, fmt.Errorf("unable to seek to LS counter placeholders: %w", err)
 	}
 	if err := writeFunc(fmt.Sprintf("outputWithMetadata counter %d", outputWithMetadataCounter), outputWithMetadataCounter); err != nil {
-		return err
+		return nil, err
 	}
 	header.OutputWithMetadataCount = outputWithMetadataCounter
 
-	return nil
+	return header, nil
 }
 
-// NewLedgerOutputWithMetadataProducer returns a OutputWithMetadataProducerFunc that provide OutputWithMetadatas from the ledger.
-func NewLedgerOutputWithMetadataProducer(l *ledger.Ledger) OutputWithMetadataProducerFunc {
-	prodChan := make(chan interface{})
+// NewLedgerUTXOStatesProducer returns a OutputWithMetadataProducerFunc that provide OutputWithMetadatas from the ledger.
+func NewLedgerUTXOStatesProducer(lastConfirmedEpoch epoch.Index, l *ledger.Ledger) UTXOStatesProducerFunc {
+	prodChan := make(chan *ledger.OutputWithMetadata)
 
 	go func() {
-		l.ForEachAcceptedUnSpentOutputWithMetadata(func(o *ledger.OutputWithMetadata) {
-			prodChan <- o
+		l.ForEachAcceptedUnspentOutputWithMetadata(func(o *ledger.OutputWithMetadata) {
```

Comment:
The ledgerstate to be dumped in the snapshot should be the notarization manager's one. That state has exactly that purpose: it already takes care of being enough epochs in the past, and if you'd apply the epochdiffs to it you would obtain the live ledger's one.